src/devices/cpu/h8/h83048.cpp

```diff
@@ -206,7 +207,12 @@ void h83048_device::device_add_mconfig(machine_config &config)
 
 void h83048_device::execute_set_input(int inputnum, int state)
 {
-	m_intc->set_input(inputnum, state);
+	if(inputnum == H8_INPUT_LINE_TEND0 || inputnum == H8_INPUT_LINE_TEND1)
+		m_tend_cb[inputnum - H8_INPUT_LINE_TEND0](state);
```

Comment:
Why have you got input lines piped straight to callbacks?  That’s rather suspect.

Reply:
This is taken directly from other H8 devices that already implement TEND0/TEND1 (i.e. `h83002_device`, `h83003_device`)